textpipe/doc.py

```diff
@@ -509,3 +510,46 @@ def aggregate_word_vectors(self, model_name=None, aggregation='mean', normalize=
             return numpy.var(vectors, axis=0).tolist()
         else:
             raise NotImplementedError(f'Aggregation method {aggregation} is not implemented.')
+
+    def _load_gensim_word2vec_model(self, model_file=None):
+        """
+        Loads pre-trained Gensim word2vec model, expects lower-cased training data
+        >>> model = Doc('')._load_gensim_word2vec_model('tests/models/gensim_test_nl.w2v')
+        >>> type(model)
+        <class 'gensim.models.keyedvectors.Word2VecKeyedVectors'>
+        """
+        lang = self.language if self.is_reliable_language else self.hint_language
+        if not self._gensim_vectors or lang not in self._gensim_vectors:
+            vectors = Word2Vec.load(model_file).wv
+            self._gensim_vectors[lang] = vectors
+        return self._gensim_vectors[lang]
+
+    @functools.lru_cache()
+    def generate_gensim_document_embedding(self, model_file=None):
+        """
+        Returns document embeddings generated with Gensim word2vec model.
+        >>> import numpy
+        >>> doc1 = Doc('textmining is verwant aan tekstanalyse')
+        >>> doc2 = Doc('textmining is verwant aan textmining')
+        >>> doc3 = Doc('tekstanalyse is verwant aan textmining')
+        >>> test_model_file = 'tests/models/gensim_test_nl.w2v'
+        >>> numpy.allclose(doc1.generate_gensim_document_embedding(model_file=test_model_file), \
+                           doc2.generate_gensim_document_embedding(model_file=test_model_file))
+        False
+        >>> numpy.allclose(doc1.generate_gensim_document_embedding(model_file=test_model_file), \
+                           doc3.generate_gensim_document_embedding(model_file=test_model_file))
+        True
+        """
+        if not model_file:
+            raise TextpipeMissingModelException('No Gensim word2vec model file specified.')
+        try:
+            model = self._load_gensim_word2vec_model(model_file)
+        except FileNotFoundError:
+            raise TextpipeMissingModelException(f'Gensim model file {model_file} is not available.')
+
+        weighted_word_vectors = [model[word.lower()] * count / model.vocab[word.lower()].count
+                                 for word, count in self.word_counts.items()
+                                 if word.lower() in model.vocab]
```

Comment:
Shouldn't we support word2vec embeddings that are not lowercased (i.e., should lowercasing either be an argument to the func and/or done someplace else?)